trunk/SUAVE/Components/Energy/Networks/Propulsor_Surrogate.py

```diff
@@ -0,0 +1,138 @@
+# propulsor_surrogate.py
+#
+# Created:  Mar 2017, E. Botero
+# Modified:
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+# suave imports
+import SUAVE
+
+# package imports
+import numpy as np
+from SUAVE.Components.Propulsors.Propulsor import Propulsor
+
+from SUAVE.Core import Data
+import sklearn
+from sklearn import gaussian_process
+from sklearn import neighbors
+from sklearn import svm
+
+# ----------------------------------------------------------------------
+#  Network
+# ----------------------------------------------------------------------
+
+# The input format for this should be Altitude, Mach, Throttle, Thrust, SFC
+
+class Propulsor_Surrogate(Propulsor):
+    def __defaults__(self): 
+        self.nacelle_diameter  = None
+        self.engine_length     = None
+        self.number_of_engines = None
+        self.tag               = 'network'
+        self.input_file        = None
+        self.sfc_surrogate     = None
+        self.thrust_surrogate  = None
+        self.areas             = Data()
+    
+    # manage process with a driver function
+    def evaluate_thrust(self,state):
+        
+        # Unpack the surrogate
+        sfc_surrogate = self.sfc_surrogate
+        thr_surrogate = self.thrust_surrogate
+        
+        # Unpack the conditions
+        conditions = state.conditions
+        altitude   = conditions.freestream.altitude
+        mach       = conditions.freestream.mach_number
+        throttle   = conditions.propulsion.throttle
+        
+        cond = np.hstack([altitude,mach,throttle])
+        
+        # Run the surrogate for a range of altitudes
+        data_len = len(altitude)
+        sfc = np.zeros([data_len,1])  
+        thr = np.zeros([data_len,1]) 
+        for ii,_ in enumerate(altitude):
+            sfc[ii] = sfc_surrogate.predict(np.array([altitude[ii][0],mach[ii][0],throttle[ii][0]]))    
+            thr[ii] = thr_surrogate.predict(np.array([altitude[ii][0],mach[ii][0],throttle[ii][0]]))   
+            
+        ## Test plot, run a range of throttles
```

Comment:
I think we should code a regression that contains this function rather than leave it in the method.